src/main/java/org/jvnet/hudson/tools/versionnumber/VersionNumberBuilder.java

```diff
@@ -150,6 +145,40 @@ private static Date parseDate(String dateString) {
     }
     
     /**
+     * Checks if the given string contains a valid value and returns that
+     * value again if it is valid or returns an empty string if it is not. A
+     * valid value encoded in the string must either be a (positive) number,
+     * convertible to an integer or a reference to an environment-variable in
+     * the form <code>${VARIABLE_NAME}</code> or <code>$VARIABLE_NAME</code>.
+     * @param buildNum The (user-provided) string which should either contain
+     *                 a number or a reference to an environment-variable.
+     * @return The given <a>buildNum</a> if valid or an empty string.
+     */
+    private static String makeValid(String buildNum) {
+        try {
```

Comment:
The constructor of `Integer` would take care of `null` as it will through an exception, I assume. But of course, being more explicit here makes sense.
Also, for trimming (which is a very good idea) it probably makes sense to do this before.

I will do as you suggest.